provider/openstack/networking.go

```diff
@@ -246,12 +219,12 @@ func getExternalNeutronNetworksByAZ(e *Environ, azName string) ([]string, error)
 	netIds := make([]string, 0)
 	for _, network := range networks {
 		for _, netAZ := range network.AvailabilityZones {
-			if azName == netAZ {
+			if azNames.Contains(netAZ) {
 				netIds = append(netIds, network.Id)
 				break
 			}
 		}
-		if azName == "" && len(network.AvailabilityZones) == 0 {
+		if azNames.IsEmpty() || len(network.AvailabilityZones) == 0 {
```

Comment:
Let's log out that we including this network because of the absence of AZ specificity. Users wondering at this behaviour will have something to go on.